Uninitialized variable build warning on tests/kernel/pipe/pipe_api

**Describe the bug**
gcc throws an unitialiazed variable build warning in tests/kernel/pipe/pipe_api (see long below)
when build with no optimizations

**To Reproduce**
Steps to reproduce the behavior:
1. twister -p nrf52_bsim -T tests/kernel/pipe/pipe_api
4. See error

OR
1. cmake -GNinja -DBOARD=native_sim ../tests/kernel/pipe/pipe_api/ -DCONFIG_NO_OPTIMIZATIONS=y
2. ninja
3. See error

**Expected behavior**
No build warnings

**Impact**
Twister failures

**Logs and console output**
```
tests/kernel/pipe/pipe_api/src/concurrency.c:34:22: error: ‘<unknown>’ may be used uninitialized [-Werror=maybe-uninitialized]
   34 |         zassert_true(k_pipe_write((struct k_pipe *)arg1, garbage, sizeof(garbage),
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   35 |                 K_MSEC(partial_wait_time)) == sizeof(garbage), "Failed to write to pipe");
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~
subsys/testsuite/ztest/include/zephyr/ztest_assert.h:180:35: note: in definition of macro ‘_zassert_base’
  180 |                         z_zassert(cond, _msg ? ("(" default_msg ")") : (default_msg), __FILE__,    \
      |                                   ^~~~
subsys/testsuite/ztest/include/zephyr/ztest_assert.h:194:9: note: in expansion of macro ‘_zassert_va’
  194 |         _zassert_va(cond, default_msg, COND_CODE_1(__VA_OPT__(1), (__VA_ARGS__), (NULL)))
      |         ^~~~~~~~~~~
subsys/testsuite/ztest/include/zephyr/ztest_assert.h:275:33: note: in expansion of macro ‘zassert’
  275 | #define zassert_true(cond, ...) zassert(cond, #cond " is false", ##__VA_ARGS__)
      |                                 ^~~~~~~
tests/kernel/pipe/pipe_api/src/concurrency.c:34:9: note: in expansion of macro ‘zassert_true’
   34 |         zassert_true(k_pipe_write((struct k_pipe *)arg1, garbage, sizeof(garbage),
      |         ^~~~~~~~~~~~
In file included from include/zephyr/kernel.h:6443,
                 from tests/kernel/pipe/pipe_api/src/concurrency.c:7:
zephyr/include/generated/zephyr/syscalls/kernel.h:1467:19: note: by argument 2 of type ‘const uint8_t *’ {aka ‘const unsigned char *’} to ‘k_pipe_write’ declared here
 1467 | static inline int k_pipe_write(struct k_pipe * pipe, const uint8_t * data, size_t len, k_timeout_t timeout)
      |                   ^~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**

 - OS: Ubuntu Linux 24.04
 - Toolchain (e.g Zephyr SDK, ...): Host gcc 14.02
 - Commit SHA or Version used: Main as of now a238e5c29d7fcef8a4ea8b4a230dd1976096c3da

**Additional context**

--- a/tests/kernel/pipe/pipe_api/src/concurrency.c
+++ b/tests/kernel/pipe/pipe_api/src/concurrency.c
@@ -13,7 +13,7 @@ LOG_MODULE_REGISTER(k_pipe_concurrency, LOG_LEVEL_DBG);
 ZTEST_SUITE(k_pipe_concurrency, NULL, NULL, NULL, NULL, NULL);
 
 static const int partial_wait_time = 2000;
-static const int dummy_data_size = 16;
+#define DUMMY_DATA_SIZE 16
 static struct k_thread thread;
 static K_THREAD_STACK_DEFINE(stack, 1024);
 
@@ -29,7 +29,7 @@ static void thread_reset(void *arg1, void *arg2, void *arg3)
 
 static void thread_write(void *arg1, void *arg2, void *arg3)
 {
-	uint8_t garbage[dummy_data_size];
+	uint8_t garbage[DUMMY_DATA_SIZE] = {};
 
 	zassert_true(k_pipe_write((struct k_pipe *)arg1, garbage, sizeof(garbage),
 		K_MSEC(partial_wait_time)) == sizeof(garbage), "Failed to write to pipe");
@@ -37,7 +37,7 @@ static void thread_write(void *arg1, void *arg2, void *arg3)
 
 static void thread_read(void *arg1, void *arg2, void *arg3)
 {
-	uint8_t garbage[dummy_data_size];
+	uint8_t garbage[DUMMY_DATA_SIZE];
 
 	zassert_true(k_pipe_read((struct k_pipe *)arg1, garbage, sizeof(garbage),
 		K_MSEC(partial_wait_time)) == sizeof(garbage), "Failed to read from pipe");
@@ -47,7 +47,7 @@ ZTEST(k_pipe_concurrency, test_close_on_read)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
 	uint8_t res;
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));
@@ -65,8 +65,8 @@ ZTEST(k_pipe_concurrency, test_close_on_write)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
-	uint8_t garbage[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
+	uint8_t garbage[DUMMY_DATA_SIZE];
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));
 	zassert_true(sizeof(garbage) == k_pipe_write(&pipe, garbage, sizeof(garbage), K_MSEC(1000)),
@@ -86,7 +86,7 @@ ZTEST(k_pipe_concurrency, test_reset_on_read)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
 	uint8_t res;
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));
@@ -107,8 +107,8 @@ ZTEST(k_pipe_concurrency, test_reset_on_write)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
-	uint8_t garbage[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
+	uint8_t garbage[DUMMY_DATA_SIZE];
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));
 	zassert_true(sizeof(garbage) == k_pipe_write(&pipe, garbage, sizeof(garbage), K_MSEC(1000)),
@@ -130,8 +130,8 @@ ZTEST(k_pipe_concurrency, test_partial_read)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
-	uint8_t garbage[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
+	uint8_t garbage[DUMMY_DATA_SIZE];
 	size_t write_size = sizeof(garbage)/2;
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));
@@ -150,8 +150,8 @@ ZTEST(k_pipe_concurrency, test_partial_write)
 {
 	k_tid_t tid;
 	struct k_pipe pipe;
-	uint8_t buffer[dummy_data_size];
-	uint8_t garbage[dummy_data_size];
+	uint8_t buffer[DUMMY_DATA_SIZE];
+	uint8_t garbage[DUMMY_DATA_SIZE];
 	size_t read_size = sizeof(garbage)/2;
 
 	k_pipe_init(&pipe, buffer, sizeof(buffer));

